### PR TITLE
Productors categorisation

### PR DESCRIPTION
**Pull request by Floury-Fabio**
https://github.com/Mate2xo/Creeons_la_coop/issues/230

`rails conversion:set_category_of_all_suppliers` for set categories of non local productors

### CODE REVIEW

File: app/views/productors/_suppliers.html.erb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          <% if category != first_category %>
+            <li class="nav-item">
+              <a class="nav-link" data-toggle="tab" href="#<%= category %>">
+                <%= t "enumerize.productor.category.#{category}" %>
+              </a>
+            </li>
+          <% end %>
+        <% end %>
+      </ul>
+    </div>
```
Comment (Mate2xo) [inline app/views/productors/_suppliers.html.erb]: Do all loops in one unique spot, and set the `active` class on the first iteration like this:
```erb
      <ul class="nav nav-tabs justify-content-center mb-5" role="tablist"> 
        <% Productor.category.values.each.with_index do |category, index| %>
            <li class="nav-item">
              <a class="nav-link <%= 'active' if index.zero? %>" data-toggle="tab" href="#<%= category %>">
                <%= t "enumerize.productor.category.#{category}" %>
              </a>
            </li>
        <% end %> 
      </ul> 
```

File: app/views/productors/_suppliers.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            <div class="tab-pane fade show" id=<%= category %> role="tabpanel" aria-labelledby="<%= category %>-tab">
+              <div class="row cols-md-space cols-sm-space cols-xs-space">
+                <%= render partial: "productors",
+                  locals: { productors: suppliers.select {|productor|
+                  productor.category == category 
+                }} %>
+              </div>
+            </div>
+          <% end %>
+        <% end %>
```
Comment (Mate2xo) [inline app/views/productors/_suppliers.html.erb]: Same here, use the index given to you by `.with_index`

File: db/schema.rb [hunk 3]
```diff
@@ -239,6 +242,16 @@
...
+  create_table "temp_enrollments_from_csv", id: false, force: :cascade do |t|
+    t.bigint "member_id"
+    t.bigint "mission_id"
+    t.bigint "id"
+    t.time "old_start_time"
+    t.time "old_end_time"
+    t.datetime "start_time"
+    t.datetime "end_time"
+  end
+
```
Comment (Mate2xo) [inline db/schema.rb]: Leak from another branch

File: lib/tasks/set_category_of_all_suppliers_to_bio.rake [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        raise ActiveRecord::Rollback unless supplier.update(category: 'bio')
+
+        puts "category of #{supplier.name} is setted"
+      end
+    end
+
+    puts ''.ljust 80, '*'
+    puts 'migration done'.center 80
+    puts ''.ljust 80, '*'
+  end
```
Comment (Mate2xo) [inline lib/tasks/set_category_of_all_suppliers_to_bio.rake:22]: Why? Isn't the `local` attribute different from the `category`?

**Review by Mate2xo — CHANGES_REQUESTED** [review]

File: lib/tasks/set_category_of_all_suppliers_to_bio.rake (hunk 4 above)
Comment (Floury-Fabio) [reply lib/tasks/set_category_of_all_suppliers_to_bio.rake:22]: Only the non local productors have a category.
If a non local productor have not a category, he doesn't appear in index.

Comment (Mate2xo) [reply lib/tasks/set_category_of_all_suppliers_to_bio.rake:22]: OK perfect, thx

**Review by Mate2xo — APPROVED** [review]

**Pull request merged**